plugin.video.npogemist/addon.py

```diff
@@ -0,0 +1,376 @@
+        "icon": item.get("thumb", DEFAULT_IMAGE),
+        "fanart": item.get("fanart", DEFAULT_FANART)
+    }
+    list_item.setArt(art)
+
+    info = {
+        "title": label,
+        "sorttitle": sorttitle,
+        "plot": item.get("plot", ""),
+        "genre": item.get("genre", ""),
+        "tagline": item.get("tagline", "")
+    }
+    if "aired" in item:
+        info["dateadded"] = dateconversions.time2kodi_datetime(item["aired"])
+    list_item.setInfo("video", info)
+
+    if "duration" in item:
+        stream = {"duration": item["duration"]}
+        list_item.addStreamInfo("video", stream)
+
+    if item.get("is_video", False):
+        list_item.setProperty("IsPlayable", "true")
+    return list_item
+
+
+def display_broadcasts(broadcasts, add_time=False):
+    """Display a list of broadcasts."""
+    for broadcast in broadcasts:
+        directory.add(router.make_url(play, mid=broadcast["mid"]), make_list_item(broadcast, add_time=add_time),
+                      is_folder=False)
+    directory.display(DATE_DURATION_TITLE)
+
+
+def display_programs(programs, sortable=True):
+    """Display a list of programs."""
+    for program_item in programs:
+        directory.add(router.make_url(program, mid=program_item["mid"]), make_list_item(program_item))
+    sort_methods = []
+    if sortable:
+        sort_methods.append(xbmcplugin.SORT_METHOD_VIDEO_TITLE)
+    directory.display(sort_methods)
+
+
+def display_mixed(items):
+    """Display a list of episodes and programs, mixed."""
+    for item in items:
+        if item.get("is_video", False):
+            directory.add(router.make_url(play, mid=item["mid"]), make_list_item(item),
+                          is_folder=False)
+        else:
+            directory.add(router.make_url(program, mid=item["mid"]), make_list_item(item))
+    sort_methods = DATE_DURATION_TITLE
+    directory.display(sort_methods)
+
+
+@router.route("/play")
+def play(mid):
+    """Play a video with id mid."""
+    # store recent
+    # TODO If we don't have a serie, we want to store the episode, but we need to adapt recents for that.
+    # For now, skip if no series.
+    try:
+        serie_id = npo.get_serie_id_by_episode(mid)
+    except NPOContentError:
+        serie_id = None
+    if serie_id is not None:
+        if serie_id in profile_recent_programs:
+            profile_recent_programs.remove(serie_id)
+        profile_recent_programs.append(serie_id)
+        # store
+        profile.set_data("recent_programs", profile_recent_programs)
+    # now play
+    try:
+        play_item = xbmcgui.ListItem(path=(npo.get_play_url(mid)))
+        # Add subtitles
+        subs_url = npo.get_subtitles_url(mid)
+        play_item.setSubtitles([subs_url])
+        xbmcplugin.setResolvedUrl(handle, True, listitem=play_item)
+        # Wait for the player to play
+        player = playing_player()
+        # Enable/disable the subtitles
+        player.showSubtitles(setting_subtitles)
```

Comment:
The `xbmc.Player` interface shouldn't be accessed in plugins (only in scripts). Adding the subtitles (`play_item.setSubtitles(...)`) only if `setting_subtitles` is true should have the same result.

Reply:
See above--the result would not be the same.

Reply:
this discussion has given me an idea how things can be done a bit nicer now.

https://github.com/Opvolger/plugin.video.uzg/commit/b2e7cd770c2b75f28bc7dd0f19da3d543bbc3dc0